core/storage/plugin_unix.go

```diff
@@ -0,0 +1,46 @@
+// +build linux freebsd solaris openbsd darwin
+
+package storage
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/go-connections/sockets"
+	"github.com/pkg/errors"
+	"github.com/rancher/agent/model"
+	"net/http"
+)
+
+func CallVolumeDriverAttach(volume model.Volume) error {
+	transport := new(http.Transport)
+	sockets.ConfigureTransport(transport, "unix", driverSockFile(volume))
+	client := &http.Client{
+		Transport: transport,
+	}
+	url := "http://volume-plugin/VolumeDriver.Attach"
+
+	bs, err := json.Marshal(struct{ Name string }{Name: volume.Name})
+	if err != nil {
+		return errors.Wrap(err, "Failed to marshal JSON")
+	}
+
+	driver := volume.Data.Fields.Driver
+	resp, err := client.Post(url, "application/json", bytes.NewReader(bs))
+	if err != nil {
+		logrus.Warnf("Failed to call /VolumeDriver.Attach '%s' (driver '%s'): %s", volume.Name, driver, err)
+		return nil
```

Comment:
In fact, the existing rancher-storage driver shouldn't have socket file in this directory, so we shouldn't fail here in any case now. So error should be returned instead of nil.